feat: loading characters from db at load and runtime (conflicts resolved)

odilitime opened the pull request:
Originally #551, this is a resubmission

<!-- Use this template by filling in information and copy and pasting relevant items out of the html comments. -->

# Relates to:
Loading Characters from db(sqlite/postgres) during runtime (via a REST API call) and
during start/load time.

<!-- LINK TO ISSUE OR TICKET -->

<!-- This risks section is to be filled out before final review and merge. -->

# Risks
_Medium_
None of these changes will affect existing workflow as its handled by env variables
FETCH_FROM_DB                            (bool) - default: null/false - For loading characters from DB
AGENT_RUNTIME_MANAGEMENT (bool) - default: null/false - For loading characters from Db at runtime

From security standpoint, all character specific secrets (EG: TWITTER_PASSWORD or OPENAI_API_KEY) is AES-256 encrypted and stored in DB

<!--
Low, medium, large. List what kind of risks, and what could be effected.
-->

# Background
For a production ready - multi-character Eliza setup, we want to load new characters in runtime via API calls.
We do not want to restart the Eliza server for each new character.

_ASSUMPTION:_ 
An api or script exists to store the characters in DB.
The api uses the same stringToUUID method to create consistent UUIDs for a character and store in DB.

## What does this PR do?
- Allows loading characters from DB during start
- Allows loading characters via REST API from DB during runtime
- The characters are stored in TEXT/JSONB format - similar to the character Agent file in sqlite and postgres respectively
- Securely encrypts each character's secrets using AES-256 encryption and then stores in DB (Decrypt after fetch)
- Proper error handling for all scenarios 
    - Character already loaded in runtime
    - Character does not exist in db

## What kind of change is this?
This is a new Feature

Why?
Currently 
1) For adding any new agent, we need to restart the Eliza server => All the agents and its clients are loaded again - All previous tweets (incase of twitter client), interactions are processed again. Any existing direct or telegram conversation is lost. 
2) Not a straight-forward mechanism to add new agents - Now with a REST API - load new agents.

Why are we doing this?
To take a step towards multi-character production setup

### Code Changes made and why?
1. **Created a new table in postgres and sqlite** - Added in the seed file/script of both DBs
`
export type CharacterTable = {
    id: UUID;                                                // created from stringToUUID - unique and consistent for name
    name: string;
    characterState: Character;                 // A JSONB or TEXT - that maps the character
    secretsIV?: Secrets;                            // Initialisation Vector for each secrets
};
`
**Also added the above in packages/core/src/types.ts**
2. **in SqliteAdapter and PostgresAdapter** - created a new function to load from this characters table
3. **in Agents/src/index.ts** -> 
- Assign Directclient to a global variable - along with set and get methods. This is to allow the same direct client be used for character(agent) runtime creation
- A function loadCharactersFromDb loadCharactersFromDb(
                            characterNames?: string
                        ): Promise<Character[]> 
    - if any characterNames argument received, it fetches only those characters from db
    - else fetches all characters
    - This handles encryption and decryption of secrets if secrets are present in a character
    - uses env.ENCRYPTION_KEY
- An express server
   - The server starts only when the env AGENT_RUNTIME_MANAGEMENT == true
   - _Why not use the same express server in DirectClient?_ DirectClient does not have access to the DB methods and all the agent-specific handling methods
   - ENDPOINT:  "/load/:agentName" METHOD: Post
   - PORT: default. - 3001, overwritten by env AGENT_PORT
4. **in packages/client-direct/src/index.ts**
- Added ENDPOINT:  "/load/:agentName" METHOD: Post
   - This endpoint (is a proxy) that routes request to the agent server's route
   -    Before proxying, checks if AGENT_RUNTIME_MANAGEMENT  ==true and if agents already in runtime
5. created a file **packages/core/src/crypt.ts**
- This handles the encryption and decryption methods
6. created scripts that parses a character json file or all files in a folder and inserts in DB
- Location: scripts/importCharactersInDB/[postgres/sqlite]/insertInDb.js
Requires env variable
`
POSTGRES_URL= # if postgres
ENCRYPTION_KEY=""
INPUT_PATH=characters # the folder path to load the characters from
SQLITE_DB_PATH= #if you want to change db path Default: agent/data/db.sqlite
`


<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->

<!-- This "Why" section is most relevant if there is no linked issue explaining why. If there is a related issue it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->

# Documentation changes needed?
Not needed necessarily.
New ENV variables and explanations are added in .env.example

<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If a docs change is needed: I have updated the documentation accordingly.
-->

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested, and probably help the PR get merged quicker. -->

# Testing
I have tested the scenarios in detailed testing steps in both sqlite and postgres.

## Where should a reviewer start?
- agent/src/index.js 
- Line 418 and 531
packages/client-direct/src/index.js
- Line 271

## Detailed testing steps
**INIT STEP:** 
1. creating character table(use the schema.sql or sqliteTables.ts ) 
2. loading characters in DB (used the above mentioned scripts in `scripts/importCharactersInDB/[postgres/sqlite]/insertInDb.js`  - 
 - also added to the characters tate and trump - TWITTER_USERNAME, TWITTER_PASSWORD, TWITTER_EMAIL to its settings.secrets and twitter to client) to test secrets encryption and decryption

**I have tested the following scenarios**
1. Fetching from database during start
- set env FETCH_FROM_DB=true
- `pnpm start`
- Will function as usual
- if we want to test postgres, set env POSTGRES_URL=''
2. Loading an agent during runtime
- set env FETCH_FROM_DB=false #if true, we can't test load as all characters in db will be loaded
- set env AGENT_RUNTIME_MANAGEMENT=true
- set env ENCRYPTION_KEY= #use the same encryption key used in insertInDB.js script
- set env AGENT_PORT=4000 #can be empty if we want to pick default port 3001
`pnpm start`
- This will load with the default character Eliza
- `curl --location --request POST 'http://localhost:3000/load/trump'` - replace port and character name if using different characters
- SUB SCENARIO 1: agent is loaded and we get success code 200
- SUB SCENARIO 2: agent is already loaded and we get error code 409
- SUB SCENARIO 3: agent does not exist in db and we get error code 404
- SUB SCENARIO 4:  Error during agent load - like incorrect twitter - error code 500
-  if we want to test postgres, set env POSTGRES_URL=''

<!--
None, automtated tests are fine.
-->

<!--
- As [anon/admin], go to [link]
  - [do action]
  - verify [result]
-->

<!-- If there is a UI change, please include before and after screenshots or videos. This will speed up PRs being merged. It is extra nice to annotate screenshots with arrows or boxes pointing out the differences. -->
<!--
## Screenshots
### Before
### After
-->

<!-- If there is anything about the deploy, please make a note. -->
<!--
# Deploy Notes
-->

<!--  Copy and paste commandline output. -->
<!--
## Database changes
-->

<!--  If there is something more than the automated steps, please specifiy deploy instructions. -->
<!--
## Deployment instructions
-->

<!-- If you are on Discord, please join https://discord.gg/ai16z and state your Discord username here for contribute role and join us in #development-feed -->
<!--
## Discord username
varkrishin

-->

odilitime committed:
update lockfile for PR1135

shakkernerd committed:
chore: remove comment

shakkernerd committed:
Merge pull request #1143 from ai16z/remove-comment

chore: remove comment

shakkernerd committed:
fix: remove unused import

shakkernerd committed:
fix: Expected an assignment or function call and instead saw an expression

shakkernerd committed:
fix: use of let instead of const

shakkernerd committed:
chore: comment out unused imports

shakkernerd committed:
fix: remove unused variable

shakkernerd committed:
chore: lint command

shakkernerd committed:
chore: add eslint config

shakkernerd committed:
chore: delete client-whatsapp folder

mcp97 committed:
Merge pull request #1139 from rarepepi/docker-fixes

fix: remove docker compose command since Docker file already runs

odilitime committed:
improve on fomo plugin and distingush it from pump.fun's plugin

shakkernerd committed:
Merge pull request #1147 from odilitime/fix-lint

fix: improve fomo integration

odilitime committed:
include fomo

odilitime committed:
update URLs and example response

odilitime committed:
Merge branch 'develop' of https://github.com/ai16z/eliza into fix-lint

shakkernerd committed:
Merge pull request #1148 from odilitime/fix-lint

chore: fix PR #1147

odilitime committed:
Merge pull request #1144 from ai16z/develop

chore: Merge monday, merging develop into main

actions-user committed:
chore: update changelog

shakkernerd committed:
feat: update packages version script

jzvikart committed:
Merge remote-tracking branch 'ai16z/develop' into realitySpiral/integration-tests

shakkernerd committed:
Merge pull request #1150 from ai16z/update-version

feat: update packages version script

jzvikart committed:
Remove dependency zx

shakkernerd committed:
chore: bump version to 0.1.6-alpha.3

shakkernerd committed:
Merge pull request #1152 from ai16z/new_version

chore: bump version to 0.1.6-alpha.3

jzvikart committed:
Minor cleanup

shakkernerd committed:
fix: fetch log level to debug

shakkernerd committed:
Merge pull request #1153 from ai16z/fetch_logs_debug

fix: fetch log level to debug

odilitime committed:
fix directClient

mcp97 committed:
Merge pull request #1154 from odilitime/fix-lint

fix: fix direct-client ability to start agents

shakkernerd committed:
Merge pull request #1155 from ai16z/develop

chore: develop into main

actions-user committed:
chore: update changelog

jzvikart committed:
Disable logging of fetch requests to avoid disclosing sensitive information

madjin committed:
update community docs

tcm390 committed:
allow multiple bots to join the voice channel

tcm390 committed:
allow multiple bots to join the voice channel

shakkernerd committed:
chore: fix broken pnpm lockfile

shakkernerd committed:
fix: client twitter login and auth handler

shakkernerd committed:
Merge pull request #1158 from ai16z/fix/client-twitter

fix: client twitter login and auth handler

yang-han committed:
fix: print commands to start the client and remove unused --non-iteractive in dockerfile

yang-han committed:
fix: print commands to start the client and remove unused --non-iteractive in dockerfile

shakkernerd committed:
chore: bump version to 0.1.6-alpha.4

shakkernerd committed:
Merge pull request #1159 from ai16z/new_version

chore: bump version to 0.1.6-alpha.4

actions-user committed:
chore: update changelog

shakkernerd committed:
Merge pull request #1156 from ai16z/tcm-discord-multiple-bot

fix: Enable multiple bots to join Discord voice channels

mcp97 committed:
Merge pull request #1163 from yang-han/fix_client

chore: print commands to start the client and remove unused --non-itera…

shakkernerd committed:
feat: make script dash compatible

shakkernerd committed:
Merge pull request #1165 from ai16z/fix/start_script

feat: make script dash compatible

nicky-ru committed:
fix: add lint script for plugin evm and fix lint errors

thomasWos committed:
Fix typo

YoungPhlo committed:
docs: Update "What Did You Get Done This Week? 5" spaces notes

jzvikart committed:
Change CI trigger to 'pull_request'

odilitime committed:
Merge branch 'develop' into main

odilitime committed:
Merge pull request #1170 from thomasWos/main

fix: Fix typo in multiversx plugin prompt for creating token

odilitime committed:
Merge branch 'develop' into docs/community-streams

odilitime committed:
Merge pull request #1174 from YoungPhlo/docs/community-streams

docs: Update "What Did You Get Done This Week? 5" spaces notes

SumeetChougule committed:
Fix client.push issue and update README for Slack client verification

ai16z-demirix committed:
test: adding tests for runtime.ts. Modified README since we switched to vitest

Commit:
fixed CONTRIBUTING.md file Issue: 1048

madjin committed:
Merge branch 'develop' of github.com:ai16z/eliza into new_docs

mcp97 committed:
Merge pull request #1191 from ileana-pr/main

docs: fixed CONTRIBUTING.md file Issue: 1048

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1190 from ai16z-demirix/main

test: adding tests for runtime.ts. Modified README since we switched to vitest

tomguluson92 committed:
Update README_CN.md

Add more details in the CN version of README

9547 committed:
docs(cn): add python2.7

Signed-off-by: 9547 <29431502+9547@users.noreply.github.com>

9547 committed:
docs(cn): rm duplicated -

Signed-off-by: 9547 <29431502+9547@users.noreply.github.com>

jzvikart committed:
Merge remote-tracking branch 'ai16z/develop' into jure/integration-tests

netdragonx committed:
Fix: optional chaining on search to prevent error when search is uncommented but not enabled in env file.

tobbelobb committed:
Write summary file before trying to cache it

 - Also give a .md file extension for prettier rendering in Discord

v1xingyue committed:
gitpod cicd bug

Sometimes we can't fetch tags, so let's fetch tags before.

shakkernerd committed:
Merge pull request #1177 from Sifchain/jure/integration-tests

feat: integration tests fixes + library improvements

marcNY committed:
Update README.md

Fix starter script: Add missing ‘cd’ command to navigate to ‘eliza-starter’ directory

Commit:
Delete scripts/gettweets.mjs

odilitime committed:
Merge pull request #1201 from 9547/doc-cn

docs(cn): add python 3.7

odilitime committed:
Merge branch 'develop' into main

odilitime committed:
Merge pull request #1207 from v1xingyue/v1xingyue-patch-1

fix: gitpod cicd bug

odilitime committed:
Merge branch 'develop' into patch-1

odilitime committed:
Merge pull request #1209 from marcNY/patch-1

docs: Update README.md

odilitime committed:
add missing change directories

odilitime committed:
Merge branch 'develop' into patch-1

odilitime committed:
Merge pull request #1196 from tomguluson92/patch-1

docs: Update "CN README" with more details

odilitime committed:
Merge branch 'ungate/dev' of https://github.com/Ungate-Ai/chitti into Ungate-Ai-ungate/dev

odilitime committed:
Merge branch 'Ungate-Ai-ungate/dev' into develop

odilitime committed:
fix bad merge

SumeetChougule committed:
fixed gitignore

madjin committed:
add wip docs

SumeetChougule committed:
Merge branch 'develop'

madjin committed:
Merge branch 'develop' of github.com:ai16z/eliza into new_docs

madjin committed:
update api docs

mcp97 committed:
Merge pull request #1211 from ai16z/new_docs

vpavlin committed:
fix: fail when cannot get token, add Akash to generateText switch

cygaar committed:
Improve generation prompt

Commit:
Rename gettweets-copy.mjs to gettweets.mjs

Commit:
Delete scripts/tweetextractor.mjs

cygaar committed:
Update

odilitime committed:
postgres needs the user to exist before you can add a participant

cygaar committed:
Merge pull request #1217 from cygaar/improve_twitter

fix: improve twitter post generation prompt

tcm390 committed:
support image message

tcm390 committed:
check content type

tcm390 committed:
correct content type

odilitime committed:
Merge branch 'main' into postgres-fix

tcm390 committed:
clean code

tcm390 committed:
handle http image

mcp97 committed:
Merge pull request #1220 from ai16z/tcm-telegram-image

fix: Allow the bot to post messages with images generated by the imageGenerationPlugin on Telegram.

mcp97 committed:
Merge pull request #1219 from odilitime/postgres-fix

fix: postgres needs the user to exist before you can add a participant

nicky-ru committed:
Merge branch 'develop' into develop

tomguluson92 committed:
feat: CircuitBreaker.ts

simpletrontdip committed:
feat: add parse mode, support telebot output in markdown

shakkernerd committed:
Merge branch 'develop' into patch-2

shakkernerd committed:
Merge pull request #1226 from tomguluson92/patch-2

fix: CircuitBreaker.ts

shakkernerd committed:
Merge branch 'develop' into twitter-credentials

shakkernerd committed:
Merge pull request #1218 from danbednarski/twitter-credentials

chore: clean up scripts

shakkernerd committed:
feat: include akash to handleProvider

shakkernerd committed:
Merge branch 'develop' into fix/akash

simpletrontdip committed:
Merge branch 'develop' into enhance-tele-client

shakkernerd committed:
Merge pull request #1214 from vpavlin/fix/akash

fix: fail when cannot get token, add Akash to generateText switch

shakkernerd committed:
Merge pull request #1229 from simpletrontdip/enhance-tele-client

feat: add parse mode=Markdown, enhance telegram bot output

azurwastaken committed:
fix type in french Readme

azurwastaken committed:
Update README_ES.md

azurwastaken committed:
Update README_IT.md

azurwastaken committed:
Merge branch 'develop' into fix/typo

yjshi2015 committed:
Enhance terminology in the project for clarity and inclusivity

jonathangus committed:
Fix: Sync UI Client with server port env

luisalrp committed:
Fix visibility issue github image cicd

tcm390 committed:
treat twitterShouldRespondTemplate as string

renlulu committed:
feat: make twitter retry limit a env

renlulu committed:
chore: env update

renlulu committed:
feat: make express payload limit configurable

renlulu committed:
Merge branch 'develop' into feat/env-twitter-retry

renlulu committed:
Merge branch 'develop' into feat/express-payload-config

tomguluson92 committed:
Update generation.ts

tomguluson92 committed:
Update index.ts

tomguluson92 committed:
Update .env.example

odilitime committed:
handle llama_local getTokenForProvider properly

mcp97 committed:
Merge pull request #1244 from renlulu/feat/env-twitter-retry

feat: make twitter login retry times as env

mcp97 committed:
Merge branch 'develop' into develop

mcp97 committed:
Merge pull request #1239 from jonathangus/develop

fix: Sync UI Client with server port env

mcp97 committed:
Merge branch 'develop' into fix/typo

mcp97 committed:
Merge pull request #1236 from azurwastaken/fix/typo

Update README for french, spanish and italian language

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1237 from yjshi2015/main

Update trump.character.json - Enhance terminology in the project for clarity and inclusivity

mcp97 committed:
Merge branch 'develop' into tcm-twitter-character-template-issue

mcp97 committed:
Merge branch 'develop' into feat/express-payload-config

mcp97 committed:
Merge branch 'develop' into fix-cicd-package-visibility

mcp97 committed:
Merge pull request #1243 from luisalrp/fix-cicd-package-visibility

Fix visibility issue github image cicd

mcp97 committed:
Merge pull request #1242 from ai16z/tcm-twitter-character-template-issue

fix: twitterShouldRespondTemplate Fails When Defined as a String in JSON Character Config

mcp97 committed:
Merge branch 'develop' into feat/express-payload-config

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1202 from netdragonx/main

fix: optional chaining on search to avoid startup errors when search is not enabled

mcp97 committed:
Merge branch 'develop' into feat/express-payload-config

mcp97 committed:
Merge pull request #1245 from renlulu/feat/express-payload-config

feat: make express payload limit configurable

mcp97 committed:
Merge branch 'develop' into fix-ll

mcp97 committed:
Merge pull request #1250 from odilitime/fix-ll

fix: Fix local_llama key warning

lalalune committed:
Update trump.character.json

Remove a controversial line so we don't have to waste time on pointless old world political divisions

julienbrs committed:
doc: add Twitter automation label notice (#1253)

mcp97 committed:
Merge pull request #1254 from julienbrs/1253--add-twitter-automation-label-notice

mcp97 committed:
Merge pull request #1252 from ai16z/lalalune-patch-1

Commit:
mapping claude vertex

shakkernerd committed:
Merge pull request #1258 from ai16z/tcm-claude-vertext-miss

fix: unsupported model provider: claude_vertex

odilitime committed:
Merge pull request #1246 from tomguluson92/tomguluson92-patch-2

feat: upgrade Tavily API with comprehensive input and constrain the token consumption

derRizzMeister committed:
feat: add README_DE.md in docs directory

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1262 from derRizzMeister/main

feat: add README_DE.md in docs directory

HashWarlock commented:
I like this a lot. In the current design, are characters able to be published to the DB through an API call? I can imagine an interesting way to leverage this is to have a published docker image and pair it with a docker compose file to load the character file into the DB on startup and then initialize the agent without needing to build a new image. 

If I am going to test this and want to load a character file, can I make a call to an S3 API storing a character file and have that be fetched and loaded in the DB? Then have my command to be something like`pnpm start --character=https://s3-bucket/folder/to/chillguy.character.json`

odilitime commented:
not yet but I like that direction. The issue with the current REST loading is that a crash will lose all your current agents and you have to know which ones to reload, so I like this DB direction. This PR is a bit over engineered, thinking about merging and then cleaning up.

odilitime committed:
Merge branch 'develop' into develop

tobbelobb committed:
Ignore agent content

oxSaturn committed:
fix: fix ENABLE_ACTION_PROCESSING logic

shakkernerd committed:
fix: pnpm lockfile

shakkernerd committed:
Merge pull request #1273 from ai16z/fix/pnpm-lockfile

fix: pnpm lockfile

shakkernerd committed:
Revert "fix: pnpm lockfile"

shakkernerd committed:
Merge pull request #1275 from ai16z/revert-1273-fix/pnpm-lockfile

chore: Revert "fix: pnpm lockfile"

shakkernerd committed:
feat: init adapter base files

shakkernerd committed:
chore: install ioredis package

shakkernerd committed:
feat: redis client base class

shakkernerd committed:
feat: add redis url param

shakkernerd committed:
chore: add ioredis types

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: console -> elizaLogger

odilitime committed:
fix build phase when pnpm-lock is cleared

ChristopherTrimboli committed:
remove TWITTER_COOKIES env var

odilitime committed:
Merge pull request #1182 from SumeetChougule/main

fix: Fix client.push issue and update README for Slack client verification

odilitime committed:
Merge branch 'develop' into main

odilitime committed:
Merge pull request #1205 from tobbelobb/main

fix: write summary file before trying to cache it

odilitime committed:
Merge branch 'develop' into develop

ChristopherTrimboli committed:
reset trump clients

ChristopherTrimboli committed:
reset lock file

ChristopherTrimboli committed:
Merge remote-tracking branch 'upstream/develop' into twitter-cookies

odilitime committed:
Merge branch 'develop' into fix/ENABLE_ACTION_PROCESSING

odilitime committed:
Merge pull request #1268 from oxSaturn/fix/ENABLE_ACTION_PROCESSING

fix: fix ENABLE_ACTION_PROCESSING logic

shakkernerd committed:
chore: pnpm lock file

odilitime committed:
Merge branch 'develop' into fix-ts

shakkernerd committed:
Merge pull request #1283 from odilitime/fix-ts

fix: fix lockfile

shakkernerd committed:
Merge branch 'develop' into feat/redis_adapter

shakkernerd committed:
Merge branch 'develop' into feat/redis_adapter

odilitime committed:
remove unneeded package that PR1182 added

odilitime committed:
remove slack added by PR1182

shakkernerd committed:
Merge pull request #1289 from odilitime/fix-dev

chore: clean up merged PR1168

twilwa committed:
fix integrationTests.yaml to properly use repo env vars

shakkernerd committed:
Merge branch 'develop' into feat/redis_adapter

twilwa committed:
Merge branch 'develop' of https://github.com/ai16z/eliza into develop

* 'develop' of https://github.com/ai16z/eliza:
  remove slack added by PR1182
  remove unneeded package that PR1182 added
  fix build phase when pnpm-lock is cleared

shakkernerd committed:
Merge pull request #1279 from ai16z/feat/redis_adapter

feat: Redis Cache Implementation

shakkernerd committed:
Merge branch 'develop' into fix/integration-fix

shakkernerd committed:
Merge pull request #1291 from ai16z/fix/integration-fix

fix: integration tests fix

shakkernerd committed:
fix: pnpm lock file

shakkernerd committed:
Merge pull request #1292 from ai16z/fix/pnpm-lockfile

fix: pnpm lock file

shakkernerd committed:
fix: add missing claude vertex case to handleProvider

shakkernerd committed:
Merge pull request #1293 from ai16z/patch/handle_provider

fix: add missing claude vertex case to handleProvider

twilwa committed:
output checkable variable for conditional

shakkernerd committed:
feat: types for Cache Store

shakkernerd committed:
chore: add adapter-redis package

shakkernerd committed:
feat: add cache env to env example

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
feat: add support for redis cache

shakkernerd committed:
Merge pull request #1294 from ai16z/fix/integration-tests

fix: output checkable variable for conditional

shakkernerd committed:
Merge branch 'develop' into feat/agent_redis_support

shakkernerd committed:
Merge pull request #1295 from ai16z/feat/agent_redis_support

feat: Add caching support for Redis

shakkernerd committed:
chore: bump version to 0.1.6-alpha.5

shakkernerd committed:
Merge pull request #1296 from ai16z/new_version_0.1.6-alpha5

chore: bump version to 0.1.6-alpha.5

shakkernerd committed:
Merge pull request #1290 from ai16z/develop

feat: Update main for v0.1.6-alpha.5

shakkernerd committed:
fix: remove clients from default character

shakkernerd committed:
Merge pull request #1297 from ai16z/fix/default_character

fix: remove clients from default character

shakkernerd committed:
fix: default character model to LLAMALOCAL

shakkernerd committed:
Merge pull request #1299 from ai16z/fix/default_character

fix: default character model to LLAMALOCAL

actions-user committed:
chore: update changelog

shakkernerd committed:
chore: Updated node version to 23.3.0 and pnpm version to 9.15.0

shakkernerd committed:
feat: release version 0.1.6

shakkernerd committed:
Merge pull request #1300 from ai16z/new_version_0.1.6

feat: release version 0.1.6

actions-user committed:
chore: update changelog

shakkernerd committed:
Merge pull request #1301 from ai16z/main

chore: rebase develop branch

Commit:
no token needed for gaianet

update getting tokens to reflect no key needed for gaianet. Don't throw an error here.

mcp97 committed:
Merge pull request #1306 from suicidalgoofy/main

no token needed for gaianet

mcp97 committed:
Merge branch 'develop' into develop

mcp97 committed:
Merge pull request #1171 from nicky-ru/develop

fix: add lint script for plugin evm and fix lint errors

mcp97 committed:
Merge branch 'develop' into twitter-cookies

mcp97 committed:
Merge pull request #1288 from ChristopherTrimboli/twitter-cookies

chore: remove TWITTER_COOKIES env var

oxSaturn committed:
fix: update turbo to fix "cannot find package" error

mcp97 committed:
Merge pull request #1307 from oxSaturn/fix/cannot-find-package

fix: update turbo to fix "cannot find package" error

mcp97 committed:
Merge branch 'develop' into develop

oxSaturn committed:
fix: set default value for cache store

oxSaturn committed:
fix: support google

mcp97 committed:
Merge pull request #1308 from oxSaturn/fix/set-default-cache-store

fix: set default value for cache store

shakkernerd committed:
chore: bump agent-twitter-client version to v0.0.17

mcp97 committed:
Merge branch 'develop' into fix/support-google

mcp97 committed:
Merge pull request #1310 from oxSaturn/fix/support-google

fix: support google model.

mcp97 committed:
Merge branch 'develop' into chore/client_twitter

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #1311 from ai16z/chore/client_twitter

chore: bump agent-twitter-client version to v0.0.17

oxSaturn committed:
fix: use MAX_TWEET_LENGTH from setting

mcp97 committed:
Merge pull request #1323 from oxSaturn/fix/max-tweet-length

fix: use MAX_TWEET_LENGTH from setting

odilitime committed:
Merge branch 'develop' of https://github.com/ai16z/eliza into develop

monilpat commented:
I would really LOVE to get this merged in for dynamic character updates
